packages/client-sdk-web/src/utils/web-client-utils.ts

```diff
@@ -7,6 +7,10 @@ export function convertToB64String(v: string | Uint8Array): string {
   return btoa(String.fromCharCode.apply(null, v));
 }
 
-export function createMetadata(cacheName: string): {cache: string} {
-  return {cache: cacheName};
+export function createMetadata(
+  cacheName: string,
+  timeoutMillis: number
+): {cache: string; deadline: string} {
+  const deadline = Date.now() + timeoutMillis;
+  return {cache: cacheName, deadline: deadline.toString()};
```

Comment:
According to [https://github.com/grpc/grpc-web#setting-deadline](url) the correct way of setting the deadline for grpc web is adding it to the metadata. Someone more familiar with pubsub should make sure that makes sense for that client.

Reply:
i think it's right